### DISCUSSION TITLE
How to make this tweak not crash instagram?

### DISCUSSION THREAD

**Discussion by EAP7-hub** [SoCuul/SCInsta#65] [Support] [answered]
Hello everyone,

I have a significant question regarding sideloading by injecting .deb files into an Instagram IPA.

With this specific Instagram hack (link to the specific tweak here: https://github.com/SoCuul/SCInsta), I am able to obtain a completely decrypted Instagram IPA. I use the version that was last tested by the developer. However, for some reason, Instagram doesn’t even load when I inject the tweak from this GitHub repository with the .deb file (com.socuul.scinsta_0.4.1_iphoneos-arm64.deb) since my iPhone is the 11 Pro and requires the arm64 version. The strange thing is that no matter what I do, the decrypted Instagram IPA file crashes, whether I inject the .deb file or not. I suspect the reason the IPA file crashes without the .deb file is that it needs to be used with the Sideloadly application for macOS. For reference, I am using Sideloadly to inject the .deb file into the Instagram IPA file.

My question is: what method should I use to inject the tweak (the .deb file) into the Instagram IPA? I have already tried METHOD 1 from the list without attempting the other options. What might I be missing? Should I enable the 2nd, 3rd, or 4th option when injecting the .deb file into the decrypted Instagram IPA?

For further reference, I obtained the decrypted Instagram IPA file from here: https://armconverter.com/decryptedappstore/us

	1.	Injecting the .deb file without any additional options in Sideloadly (no Cydia Substrate, Substitute, or Sideload Spoofer). (I HAVE ALREADY TRIED THIS OPTION, WHICH AGAIN MAKES INSTAGRAM NOT LOAD AT ALL. IT JUST CRASHES EVERY TIME I OPEN THE APP.)

Which of these options should I try to prevent Instagram from crashing? Am I missing something when trying to inject the SCInstagram tweak? I need your help!

Here are the options I’m unsure about:
2. Enabling Cydia Substrate in Sideloadly.
3. Using the Substitute option in Sideloadly.
4. Enabling Sideload Spoofer in Sideloadly.

**Comment by Matsuui** [comment]
For instagram to work while sideloaded you have to build the tweaked ipa on your computer instead of injecting with Sideloadly. There are instructions in the README if you still need.

**Comment by SoCuul** [comment, marked as answer]
Unfortunately the tweak has other prerequisites not present in the distributable debs, so as @Matsuui said the only way is to build the tweak from scratch with the instructions in the README.
